vcd/resource_vcd_vapp_vm.go

```diff
@@ -100,6 +103,25 @@ func resourceVcdVAppVm() *schema.Resource {
 				Optional: true,
 				ForceNew: true,
 			},
+			"disk": {
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+					"name": {
```

Comment:
Elephant in the room and the most important question of this review:
- Why is disk defined in VM, as opposed to VM defined in a disk? Analogy: `vcd_catalog_item` where catalog is linked inside the item.
- How can we attach more than one disk to a VM?

Reply:
Good point we can do it from both sides.
* From VM side attach I did cause it was more "natural" for me. And like this we do from UI(vpshere).
* unlimited:
```
resource "vcd_vapp_vm" "web1" {
  name = "web-01-${count.index}"
  vapp_name = "${vcd_vapp.web.name}"
  catalog_name  = "testCatalog"
  template_name = "photon-hw11"

  network_name = "TestAccVcdVAppVmNetFwTest"
  memory        = 1024
  cpus          = 1
  ip            = "allocated"
  count = 1

  disk {
    name = "${vcd_independent_disk.Disk_1.name}"
    bus_number = 1
    unit_number = 0
  }

  disk {
    name = "${vcd_independent_disk.Disk_2.name}"
    bus_number = 1
    unit_number = 1
  }

  depends_on = ["vcd_vapp.web"]
}
```

Reply:
I would vote for disk being defined in VM as it is done now. It is more natural and replicates vSphere behaviour.

Reply:
I agree. I haven't realised you can define more than one `disk { ... }` section. It's really cool! Please add this sample with two disks to documentation, as it's not clear by default.

Reply:
Done